test/WebResourceInvalidInputTest.js

```diff
@@ -0,0 +1,79 @@
+import * as helpers from './helpers';
+import * as chai from 'chai';
+import * as chaiDateTime from 'chai-datetime';
+import { registerStorageAdapter } from '../out/storage-adapters';
+import {
+	DISK_STORAGE_ADAPTER_NAME,
+	DiskStorageAdapter,
+} from './storage-adapters/disk-storage-adapter';
+
+chai.use(chaiDateTime);
+// const { expect } = chai;
+
+registerStorageAdapter(DISK_STORAGE_ADAPTER_NAME, DiskStorageAdapter);
+
+const hex = '5261777221';
+const buf = Buffer.from(hex, 'hex');
+
+const baseInput = {
+	filename: 'logo.png',
+	data: buf,
+	contentType: 'image/png',
+	contentDisposition: null,
+	size: buf.byteLength,
+	storage: DISK_STORAGE_ADAPTER_NAME,
+};
+
+helpers.describe('WebResource', (test) => {
+	describe('validate no object', () => {
+		test.validate(hex, true, new Error('is not an object: string'));
+	});
+
+	describe('validate no filename', () => {
+		const noFilename = { ...baseInput };
+		// @ts-ignore
+		delete noFilename.filename;
```

Comment:
If you want to avoid the typing issue you can do:
```suggestion
		const { filename, ...noFilename } = baseInput;
```
(and the same pattern elsewhere)

Reply:
nice! thanks